src/main/java/com/jsoniter/IterImpl.java

```diff
@@ -469,7 +470,10 @@ static final double readDouble(final JsonIterator iter) throws IOException {
                     decimalPart = -decimalPart;
                     int decimalPlaces = iter.head - start;
                     if (decimalPlaces > 0 && decimalPlaces < IterImplNumber.POW10.length && (iter.head - oldHead) < 10) {
-                        return value + (decimalPart / (double) IterImplNumber.POW10[decimalPlaces]);
+                    	BigDecimal integerPart = new BigDecimal(value);
+                    	BigDecimal fractionalPart = new BigDecimal((decimalPart / (double) IterImplNumber.POW10[decimalPlaces]));
+                    	BigDecimal result = integerPart.add(fractionalPart).setScale(decimalPlaces,BigDecimal.ROUND_HALF_UP);
+                    	return result.doubleValue();
```

Comment:
@t-mangoe do you have tests and benchmarks that prove need for this change?

also I would move this commit to the separated pull request

Reply:
@plokhotnyuk Sorry. This commit targets another issue. I will this commit, and make separeted pull request. Thanks.